animation-template/src/main.rs

```diff
@@ -1,69 +1,96 @@
-mod animation;
-
-use std::{
-    error::Error,
-    io::{BufRead, BufReader},
-};
-
-use animation_api::{JsonRpcMessage, JsonRpcMethod};
-use serde::Serialize;
+use animation_api::{AnimationParameters, StepAnimation};
+use animation_utils::animation_plugin;
+use lightfx::parameter_schema::{Parameter, ParameterValue, ParametersSchema};
+use serde::{Deserialize, Serialize};
 use serde_json::json;
 
-fn receive(
-    reader: &mut impl BufRead,
-) -> Result<Option<JsonRpcMessage<JsonRpcMethod>>, Box<dyn Error>> {
-    let mut buffer = String::new();
-    if reader.read_line(&mut buffer)? == 0 {
-        Ok(None)
-    } else {
-        Ok(Some(serde_json::from_str(&buffer)?))
-    }
+#[derive(Serialize, Deserialize)]
+struct Parameters {
+    // TODO: Define your animation's parameters
+    tail_length: usize,
 }
 
-fn respond<T>(id: Option<usize>, payload: T)
-where
-    T: Serialize,
-{
-    let Some(id) = id else { return; };
+pub struct Animation {
```

Comment:
nit: if this aims to be an example Animation rather than a generic base animation, how about noting that in the name? e.g.  `ExampleAnimation` or `MyAnimation`. I think it would help distinguish what needs to be changed and what doesn't